src/Databases/PostgreSQL/DatabaseMaterializedPostgreSQL.cpp

```diff
@@ -162,8 +249,123 @@ void DatabaseMaterializedPostgreSQL::createTable(ContextPtr local_context, const
         return;
     }
 
-    throw Exception(ErrorCodes::NOT_IMPLEMENTED,
-        "Create table query allowed only for ReplacingMergeTree engine and from synchronization thread");
+    const auto & create = query->as<ASTCreateQuery>();
+    if (!create->attach)
+        throw Exception(ErrorCodes::QUERY_NOT_ALLOWED, "CREATE TABLE is not allowed for database engine {}. Use ATTACH TABLE instead", getEngineName());
+
+    /// Create ReplacingMergeTree table.
+    auto query_copy = query->clone();
+    auto * create_query = assert_cast<ASTCreateQuery *>(query_copy.get());
+    create_query->attach = false;
+    create_query->attach_short_syntax = false;
+    DatabaseAtomic::createTable(StorageMaterializedPostgreSQL::makeNestedTableContext(local_context), table_name, table, query_copy);
+
+    /// Attach MaterializedPostgreSQL table.
+    attachTable(table_name, table, {});
+}
+
+
+void DatabaseMaterializedPostgreSQL::attachTable(const String & table_name, const StoragePtr & table, const String & relative_table_path)
+{
+    /// If there is query context then we need to attach materialized storage.
+    /// If there is no query context then we need to attach internal storage from atomic database.
+    if (CurrentThread::isInitialized() && CurrentThread::get().getQueryContext())
+    {
+        auto current_context = Context::createCopy(getContext()->getGlobalContext());
+        current_context->setInternalQuery(true);
+
+        /// We just came from createTable() and created nested table there. Add assert.
+        auto nested_table = DatabaseAtomic::tryGetTable(table_name, current_context);
+        assert(nested_table != nullptr);
+
+        try
+        {
+            auto tables_to_replicate = settings->materialized_postgresql_tables_list.value;
```

Comment:
I don't understand where the source of truth? Which tables we have in `materialized_postgresql_tables_list` which tables we have in `materialized_tables`?

Reply:
> I don't understand where the source of truth? 

Sounds so philosophically...

p.s. discussed in dm that I need to add more comments.